lib/net.js

```diff
@@ -708,13 +706,11 @@ protoGetter('remotePort', function remotePort() {
 
 Socket.prototype._getsockname = function() {
   if (!this._handle || !this._handle.getsockname) {
-    return {};
-  }
-  if (!this._sockname) {
-    const out = {};
-    const err = this._handle.getsockname(out);
-    if (err) return {};  // FIXME(bnoordhuis) Throw?
-    this._sockname = out;
+    return this._sockname || {};
```

Comment:
This does seem to change the behavior in case `this._sockname` is not an empty object?